src/Avalonia.Diagnostics/Diagnostics/VisualExtensions.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.IO;
+using Avalonia.Controls;
+using Avalonia.Media.Imaging;
+using Avalonia.VisualTree;
+
+namespace Avalonia.Diagnostics
+{
+    static class VisualExtensions
+    {
+        /// <summary>
+        /// Rendered control to stream
+        /// </summary>
+        /// <param name="source">the control I want to render in the stream</param>
+        /// <param name="destination">destination destina</param>
+        /// <param name="dpi">(optional) dpi quality default 96</param>
+        public static void RenderTo(this IControl source, Stream destination, double dpi = 96)
+        {
+            var rect = source.Bounds;
+            var pixelSize = new PixelSize((int)rect.Width, (int)rect.Height);
+            var dpiVector = new Vector(dpi, dpi);
+
+            // get Visual root
+            var root = (source.VisualRoot 
+                ?? source.GetVisualRoot())
+                as IControl ?? source;
+
+            // Backup current vaules
+            var oldgeometry = root.Clip;
+            var oldClipToBounds = root.ClipToBounds;
+            var oldRenderTransformOrigin = root.RenderTransformOrigin;
+            var oldrenderTransform = root.RenderTransform;
```

Comment:
All these properties are styles properties.
Right now you read previous value or any priority, set new value with local value priority, then set old value with local value priority. After all these properties might have invalid values and unexpected to the user results.
I.e. let's say user temporary set RenderTransform with style or animation priority, but this code reset same value as a local value which will stay there even after animation/style was removed.

To solve this problem just use SetValue with highest disposable priority - Animation:
```
IDisposable clipSetter = root.SetValue(Visual.ClipProperty, clipRegion, BindingPriority.Animation);
// ... Rendering code ...
clipSetter.Dispose();
```

Reply:
Thanks I did not know this behavior.